code/modules/mob/living/carbon/damage_procs.dm

```diff
@@ -0,0 +1,191 @@
+		take_overall_damage(0, amount)
+	else
+		heal_overall_damage(0, -amount)
+
+
+/mob/living/carbon/adjustToxLoss(amount, updating_health=1)
+	if(has_dna() && TOXINLOVER in dna.species.specflags) //damage becomes healing and healing becomes damage
+		amount = -amount
+		if(amount > 0)
+			blood_volume -= 5*amount
+		else
+			blood_volume -= amount
+	return ..()
+
+////////////////////////////////////////////
+
+//Returns a list of damaged bodyparts
+/mob/living/carbon/proc/get_damaged_bodyparts(brute, burn)
+	var/list/obj/item/bodypart/parts = list()
+	for(var/X in bodyparts)
+		var/obj/item/bodypart/BP = X
+		if((brute && BP.brute_dam) || (burn && BP.burn_dam))
+			parts += BP
+	return parts
+
+//Returns a list of damageable bodyparts
+/mob/living/carbon/proc/get_damageable_bodyparts()
+	var/list/obj/item/bodypart/parts = list()
+	for(var/X in bodyparts)
+		var/obj/item/bodypart/BP = X
+		if(BP.brute_dam + BP.burn_dam < BP.max_damage)
+			parts += BP
+	return parts
+
+//Heals ONE bodypart randomly selected from damaged ones.
+//It automatically updates damage overlays if necessary
+//It automatically updates health status
+/mob/living/carbon/heal_bodypart_damage(brute, burn, only_robotic = 0, only_organic = 1)
+	var/list/obj/item/bodypart/parts = get_damaged_bodyparts(brute,burn)
+	if(!parts.len)
+		return
+	var/obj/item/bodypart/picked = pick(parts)
+	if(picked.heal_damage(brute, burn, only_robotic, only_organic))
+		update_damage_overlays()
+
+//Damages ONE bodypart randomly selected from damagable ones.
+//It automatically updates damage overlays if necessary
+//It automatically updates health status
+/mob/living/carbon/take_bodypart_damage(brute, burn)
+	var/list/obj/item/bodypart/parts = get_damageable_bodyparts()
+	if(!parts.len)
+		return
+	var/obj/item/bodypart/picked = pick(parts)
+	if(picked.take_damage(brute,burn))
+		update_damage_overlays()
+
+
+//Heal MANY bodyparts, in random order
+/mob/living/carbon/heal_overall_damage(brute, burn, only_robotic = 0, only_organic = 1, updating_health = 1)
+	var/list/obj/item/bodypart/parts = get_damaged_bodyparts(brute,burn)
+
+	var/update = 0
+	while(parts.len && (brute>0 || burn>0) )
+		var/obj/item/bodypart/picked = pick(parts)
+
+		var/brute_was = picked.brute_dam
+		var/burn_was = picked.burn_dam
+
+		update |= picked.heal_damage(brute,burn, only_robotic, only_organic, 0)
+
+		brute -= (brute_was-picked.brute_dam)
+		burn -= (burn_was-picked.burn_dam)
+
+		parts -= picked
+	if(updating_health)
+		updatehealth()
+		if(update)
+			update_damage_overlays()
+
+// damage MANY bodyparts, in random order
+/mob/living/carbon/take_overall_damage(brute, burn, updating_health = 1)
+	if(status_flags & GODMODE)
+		return	//godmode
+
+	var/list/obj/item/bodypart/parts = get_damageable_bodyparts()
+	var/update = 0
+	while(parts.len && (brute>0 || burn>0) )
+		var/obj/item/bodypart/picked = pick(parts)
+		var/brute_per_part = round(brute/parts.len, 0.01)
+		var/burn_per_part = round(burn/parts.len, 0.01)
```

Comment:
one thing I forgot to mention: we round the dmg on each part to 0.01 so we don't get huge decimal numbers when we split some damage between each bodypart.